hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -322,7 +329,11 @@ public void persistToken(Token token) {
 				final var accountId = new EntityId(uniqueToken.getOwner());
 				currentUniqueTokens.remove(merkleUniqueTokenId);
 				currentUniqueTokenAssociations.disassociate(new EntityId(uniqueToken.getTokenId()), merkleUniqueTokenId);
-				currentUniqueOwnershipAssociations.disassociate(accountId, merkleUniqueTokenId);
+				if (uniqueToken.getOwner().equals(token.getTreasury().getId())) {
```

Comment:
We can add inline docs do specify that in the first case we are going in in `burn` and the second  case is from wipe (going to the else)